src/App.tsx

```diff
@@ -15,47 +15,91 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
 export const App: React.FC = () => {
+  const [sortType, setValue] = useState(SortType.NONE);
+  const [isReversed, setReversed] = useState(false);
+
+  const reverse = (e: React.MouseEvent<HTMLButtonElement>) => {
+    e.currentTarget.classList.toggle('is-light');
+    setReversed(!isReversed);
+  };
+
+  const visibleGoods = [...goodsFromServer];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+      default:
+        return 1;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  const isStartArray
+    = goodsFromServer.toString() === visibleGoods.toString();
+
+  const buttonReset = (
+    <button
+      type="button"
+      className="button is-danger is-light"
+      onClick={() => {
+        return (setValue(0), setReversed(false));
+      }}
+    >
+      Reset
+    </button>
+  );
+
   return (
     <div className="section content">
       <div className="buttons">
         <button
           type="button"
-          className="button is-info is-light"
+          className={`button is-info ${sortType === 1 ? '' : 'is-light'}`}
```

Comment:
```suggestion
          className={
            classNames('button is-info',
              {
                'is-light': sortType !== SortType.ALPHABET,
              })
          }
```

use classNames function if you have condition
and avoid using 'magic' numbers

Reply:
fix everywhere